sklearn/neural_network/tests/test_mlp.py

```diff
@@ -633,3 +638,213 @@ def test_n_iter_no_change_inf():
+                         sample_weight=np.hstack([sample_weight,
+                                                  sample_weight]))
+
+
+def test_sample_weight_invariance(n_samples=50):
+    # Test sample weight loss functions directly
+
+    # regression
+
+    # squared loss
+    random_state = check_random_state(0)
+    y_true = random_state.random_sample(size=(n_samples, 1))
+    y_pred = random_state.random_sample(size=(n_samples, 1))
+    check_sample_weight_invariance(name='squared_loss', metric=squared_loss,
+                                   y1=y_true, y2=y_pred)
+
+    # classification
+
+    # binary log loss
+    random_state = check_random_state(0)
+    y_true = random_state.randint(0, 2, size=(n_samples, 1))
+    y_pred = random_state.random_sample(size=(n_samples, 1))
+    check_sample_weight_invariance(name='binary_log_loss',
+                                   metric=binary_log_loss,
+                                   y1=y_true, y2=y_pred)
+
+    # log loss
+    random_state = check_random_state(0)
+    y_true = random_state.randint(0, 5, size=(n_samples, 1))
+    y_pred = random_state.random_sample(size=(n_samples, 1))
+    check_sample_weight_invariance(name='log_loss', metric=log_loss,
+                                   y1=y_true, y2=y_pred)
+
+
+def test_mlp_sample_weight():
+    # validate sample weight behaviour through fit
+    n_samples = 100
+    X = X_digits_binary[:n_samples]
+    y = y_digits_binary[:n_samples]
+    seed = 0
+
+    # validate sample weight and X have the same shape when calling fit
+    # o.w. raise meaningful message
+    sample_weight = np.ones(n_samples-1, dtype=np.float64, order='C')
+    message = ('Shapes of X and sample_weight do not match.')
+    learners = [MLPClassifier(random_state=seed),
+                MLPRegressor(random_state=seed)]
+    for learner in learners:
+        # fit
+        assert_raise_message(ValueError, message, learner.fit, X, y,
+                             sample_weight)
+        # partial fit
+        if isinstance(learner, MLPRegressor):
+            continue
+        assert_raise_message(ValueError, message, learner.partial_fit, X, y,
+                             classes=np.unique(y),
+                             sample_weight=sample_weight)
+
+    # None sample weight or all being 1 predicts same output
+    test_sample = np.ones(X.shape[1], dtype=np.float64).reshape(1, -1)
+    sample_weight = np.ones(n_samples, dtype=np.float64, order='C')
+
+    for learner in [MLPClassifier, MLPRegressor]:
+
+        predicted = (
+                    learner(random_state=seed)
+                    .fit(X, y)
+                    .predict(test_sample)
+                )
+
+        predicted_sw = (
+                    learner(random_state=seed)
+                    .fit(X, y, sample_weight)
+                    .predict(test_sample)
+                )
+
+        assert_array_equal(predicted, predicted_sw)
+
+
+def test_mlp_class_weight():
```

Comment:
There are also generic tests for class_weight in estimator_checks.py

What's missing from estimator_checks is a test that sample_weight has any (or the right) effect.

Reply:
Will check it out.

Anyway about the sample_weight effect I made the hyper plane test, where it moves by giving low weights to one class.

And another test is the following:
```
    # data for balance check
    X_bal = np.array([[-1.0, -1.0], [-1.0, 0], [-.8, -1.0], [1.0, 1.0]])
    y_bal = [1, 1, -1, -1]
    # class_weight = 'balanced', and class_weight = None should return
    # same values when y has equal number of all labels
    clf.fit(X_bal, y_bal, class_weight=None)
    prob = clf.predict_proba(test_sample)

    clf.fit(X_bal, y_bal, class_weight='balanced')
    prob_b = clf.predict_proba(test_sample)

    assert_equal(len(clf.classes_), 2)
    assert_array_almost_equal(prob, prob_b)

```